### PR TITLE
Improve animation startup synchronization

### PR DESCRIPTION
**Pull request by asdasdduck**
https://github.com/BabyImpala/SexLabpp/pull/14

### CODE REVIEW

File: dist/Source/Scripts/sslActorAlias.psc [hunk 1]
```diff
@@ -420,10 +417,31 @@ bool Function SetActor(Actor ProspectRef)
 	return false
 EndFunction
 
-; Take this actor out of combat and clear all actor states
-Function StartSetActorInterrupts() native
-; Undo "StartSetActorInterrupts()" persistent changes
-Function EndSetActorInterrupts() native
+Function LockActorForAnimation()
```
Comment (KrisV-777) [inline dist/Source/Scripts/sslActorAlias.psc]: LockActorForAnimation() is only called by LockActor() in State Paused, NativeActorLockApplied() is only called by LockActorForAnimation(), theres a double check for _ActorLocked and `_ActorLocked` is only set after NativeActorLockApplied. These two functions should probably be merged
Avoid putting function definitions which are exclusively intended for a specific state depentend action in the empty state, to ensure these arent called out of order

Function calls are among the slowest things you can do in Papyrus, the compiler cannot inline functions. So I would recommend moving this entire code into LockActor() in Paused Statem and remove both of these functions

File: dist/Source/Scripts/sslActorAlias.psc [hunk 2]
```diff
@@ -618,6 +612,7 @@ State Paused
 		EndIf
 		_HomoTypes = _Thread.CheckActiveHomoTypes()
 		UpdateBaseEnjoymentCalculations()
+		_Thread.AnimationStart()
```
Comment (KrisV-777) [inline dist/Source/Scripts/sslActorAlias.psc]: Why was this moved down so much? The Utility.Wait() was specifically put after AnimationStart() to ensure it doesnt slow down the thread startup. Did this cause issues?

File: dist/Source/Scripts/sslThreadModel.psc [hunk 3]
```diff
@@ -1485,7 +1532,50 @@ Function PlayStageAnimations()
 	RealignActors()
 EndFunction
 
-; Set location for all _Positions on CenterAlias, incl offset, and play their respected animation. _Positions are assumed to be sorted by scene
+Function OnAnimationSyncFailed()
```
Comment (KrisV-777) [inline dist/Source/Scripts/sslThreadModel.psc:1569]: All of these GetStatus() checks should be removed and instead move the Function into the required state directly, with the declaration in the empty state only throwing an error (see other local function definitions as an example)

**Review by KrisV-777 — COMMENTED** [review]
Just some small changes where functions should be moved or manually inlined, then we can merge in Papyrus. C++ codebase is fine as is

File: dist/Source/Scripts/sslActorAlias.psc (hunk 2 above)
Comment (asdasdduck) [reply dist/Source/Scripts/sslActorAlias.psc]: ~~The delayed SOSBend0. Otherwise it's a race issue with dispatching and can lead to animation graph abnormalities. Can test a workaround, but that just seems like a hole for rare edge case problems. ~~

Correction: Moved it further up. The Bend used an existing clip therefor it's not a race issue.

File: dist/Source/Scripts/sslActorAlias.psc [hunk 4]
```diff
@@ -417,17 +417,6 @@ bool Function SetActor(Actor ProspectRef)
...
-		return
-	EndIf
-	_ActorRef.SetActorValue("Paralysis", 0.0)
-	_ActorRef.SetFactionRank(_AnimatingFaction, 1)
-	_ActorRef.SetAnimationVariableInt("IsNPC", 0)
-	_ActorRef.SetAnimationVariableBool("bHumanoidFootIKDisable", 1)
-	NativeActorLockApplied()
-EndFunction
-
 Function NativeActorLockApplied()
```
Comment (KrisV-777) [inline dist/Source/Scripts/sslActorAlias.psc:420]: This can also be removed

Comment (asdasdduck) [reply dist/Source/Scripts/sslActorAlias.psc:420]: Remove NativeActorLockApplied? It's called in many places. I can remove/integrate LockActorForAnimation(), but the native-preparation finalization still needs a callable implementation.

Comment (KrisV-777) [reply dist/Source/Scripts/sslActorAlias.psc:420]: Ah, nvm. My VSCode didnt show other instances of this function being used
